src/ApiCompat/Microsoft.DotNet.ApiCompatibility/DefaultSymbolsEqualityComparer.cs

```diff
@@ -27,6 +27,12 @@ private static string GetKey(ISymbol symbol)
                     return method.Name;
             }
 
+            if (symbol is IAssemblySymbol assembly)
+            {
+                // When mapping assemblies, we must ignore the version, culture and token.
+                return assembly.Identity.Name;
+            }
```

Comment:
This is a very interesting change. Before, an assembly with the same identity but different version, culture or public key didn't not map together. As an example, System.Runtime/6.0.0.0 didn't map with System.Runtime/7.0.0.0.

Reply:
I'm not sure where this is used, but maybe you can test these scenarios to make sure we catch them?
1. Error when changing public key.
2. Error when version is lower on right than left.
3. Ensure we compare newer versions of the same assembly and find compatibility problems in them.
4. Either error when changing culture, or support mapping multiple assemblies with different cultures.  I'm not sure there's a scenario for API compat on culture-specific assemblies since they typically have no code in them just embedded resources.

Reply:
Yes, what you describe is already being validated by the `AssemblyIdentityMustMatch` rule here: https://github.com/dotnet/sdk/blob/c4bf096954a96e0a2be7cfcbfe0c4aacca35c038/src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AssemblyIdentityMustMatch.cs#L24-L136.

This specific change here affects the mapping of assemblies (before the rule is even invoked) and resulted in assemblies with different full qualified names (i.e. the version) not being mapped together.